site/source/tutorials/internationalization.md

```diff
@@ -0,0 +1,504 @@
+formatUnit(5280, 'foot', { form: 'narrow' }); // '5,280′'
+```
+
+## Internationalization with `@dojo/widget-core`
+
+While `@dojo/i18n` is designed to be an independent package, nearly every Dojo 2 application will have a view component. So [`@dojo/widget-core`](https://github.com/widget-core/) provides a custom mixin (`@dojo/widget-core/mixins/I18n`) to make working with `@dojo/i18n` more friendly. Widgets that have incorporated this mixin can localize message bundles by passing them to the `localizeBundle` method during rendering. If messages for the widget's locale have not been loaded yet, then the default messages are returned, and the widget is invalidated once the locale-specific messages have loaded. The object returned by the `localizeBundle` method contains all the messages, as well as a format method that takes a message key as well as any options. If message formatting is not required, then messages can be directly accessed (e.g., `messages.hello`).
+
+In addition to the `localizeBundle` method, `@dojo/widget-core/mixins/I18n` introduces two properties: a boolean `rtl` property and a string `locale` property. The boolean `rtl` property determines whether the text decoration is right-to-left (`true`) or left-to-right(`false`: the default). The `locale` property specifies the locale used to determine which message translations are rendered. Since each widget can have its own distinct locale, it is possible to use multiple languages in the same application. If no `locale` property is included, widgets assume the root application locale (`i18n.locale`). Note, however, that since Dojo 2 widgets are controlled, child widgets do not automatically inherit the locale from their parent. Parents must pass the correct locale as needed to child widgets; otherwise, a child widget will use the application locale while its parent uses a custom locale.
+
+```typescript
+import I18nMixin, { I18nProperties } from '@dojo/widget-core/mixins/I18n';
+import greetings from './nls/greetings';
+
+export interface GreetingProperties extends I18nProperties {
+	name: string;
+}
+
+const GreetingBase = I18nMixin(WidgetBase);
+export default class Greeting extends GreetingBase<GreetingProperties> {
+	render() {
+		const messages = this.localizeBundle(greetings);
+		const { name } = this.properties;
+		const hello = messages.format('hello', { name });
+
+		return v('p', [ hello ]);
+	}
+}
+```
+
+## The Build Process
+
+Dojo 2 uses [webpack](https://webpack.js.org) to bundle applications. Since Dojo 2's packages are designed to function independently, a few extra steps are required to ensure that CLDR data and locale-specific message translations are included in the build. Dojo 2's build is configured via a `.dojorc` at the project's root directory. `.dojorc` is a JSON file containing settings defined under specific namespaces. All build options are specified under the `build-webpack` namespace, and the following are specific to internationalization:
```

Comment:
Dojo 2's build -> should probably link to https://dojo.io/tutorials/006_deploying_to_production/ or add a link to getting started with a Dojo 2 build?